src/Common/tests/Collections/IDictionaryTest.cs

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Tests.Collections
+{
+    public abstract class IDictionaryTest<TKey, TValue> : ICollectionTest<KeyValuePair<TKey, TValue>>
+    {
+        protected IDictionaryTest(bool isSynchronized)
+            : base(isSynchronized)
+        {
+        }
+
+        protected IDictionary<TKey, TValue> GetDictionary(object[] items)
+        {
+            return (IDictionary<TKey, TValue>)GetCollection(items);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(16)]
+        [InlineData(100)]
+        public void KeysShouldBeCorrect(int count)
+        {
+            object[] items = GenerateItems(count);
+            IDictionary<TKey, TValue> dict = GetDictionary(items);
```

Comment:
This ```dict``` isn't used until the assert below... consider moving it down closer to where it's used.